graylog2-server/src/main/java/org/graylog2/indexer/cluster/Cluster.java

```diff
@@ -90,63 +103,54 @@ public ClusterHealthResponse health() {
      *
      * @return the cluster health response
      */
-    public ClusterHealthResponse deflectorHealth() {
-        ClusterHealthRequest request = new ClusterHealthRequest(indexSetRegistry.getWriteIndexAliases());
-        return c.admin().cluster().health(request).actionGet();
-    }
-
-    public Map<String, NodeInfo> getDataNodes() {
-        return getAllNodes().entrySet().stream()
-                .filter(n -> n.getValue().getSettings().getAsBoolean("node.data", true))
-                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    public Optional<JsonObject> deflectorHealth() {
+        return Optional.ofNullable(clusterHealth(Arrays.asList(indexSetRegistry.getWriteIndexAliases())));
     }
 
-    public Map<String, NodeInfo> getAllNodes() {
-        final ClusterAdminClient clusterAdminClient = c.admin().cluster();
-        final NodesInfoRequest request = clusterAdminClient.prepareNodesInfo()
-                .all()
-                .request();
-
-        final ImmutableMap.Builder<String, NodeInfo> builder = ImmutableMap.builder();
-        for (NodeInfo nodeInfo : clusterAdminClient.nodesInfo(request).actionGet().getNodes()) {
-            builder.put(nodeInfo.getNode().id(), nodeInfo);
-        }
-
-        return builder.build();
-    }
-
-    public Map<String, NodeStats> getNodesStats(String... nodesIds) {
-        final ClusterAdminClient clusterAdminClient = c.admin().cluster();
-        final NodesStatsRequest request = clusterAdminClient.prepareNodesStats(nodesIds).request();
-        final ImmutableMap.Builder<String, NodeStats> builder = ImmutableMap.builder();
-        for (NodeStats nodeStats : clusterAdminClient.nodesStats(request).actionGet().getNodes()) {
-            builder.put(nodeStats.getNode().id(), nodeStats);
+    /**
+     * Retrieve the response for the <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/cat-nodes.html">cat nodes</a> request from Elasticsearch.
+     *
+     * @param fields The fields to show, see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/cat-nodes.html">cat nodes API</a>.
+     * @return A {@link JsonArray} with the result of the cat nodes request.
+     */
+    public JsonArray catNodes(String fields) {
+        final Cat request = new Cat.NodesBuilder()
+                .setParameter("h", fields)
+                .setParameter("full_id", true)
+                .build();
+        final CatResult response;
+        try {
+            response = jestClient.execute(request);
+        } catch (IOException e) {
+            LOG.error("Unable to read Elasticsearch node file descriptors", e);
+            return new JsonArray();
         }
 
-        return builder.build();
+        return response.getJsonObject().getAsJsonArray("result");
```

Comment:
Should we really return a `JsonArray` here, considering it's a public method? It's not really the most intuitive data structure to work with.